.github/actions/get-changed-files/get-changed-files.sh

```
@@ -0,0 +1,80 @@
+echo "__ DIFF found __"
+echo "$DIFF"
+
+# Filter the DIFF to just the directories specified in the input files
+if [ "$FILTER" != "." ]; then
+  echo "__ filtering DIFF to only include $FILTER __"
+  FILTERED_DIFF=""
+  IFS=$'\n'
+  for file in $DIFF; do
+    while IFS= read -r pattern || [ -n "$pattern" ]; do
+      clean_pattern=${pattern%/}
+      if [[ $file == $clean_pattern || $file == $clean_pattern/* ]]; then
```

Shellcheck (0.10.0)

[warning] Quote the right-hand side of == in [[ ]] to prevent glob matching.

(SC2053)